tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -933,6 +933,37 @@ func (az *AzureProvider) destroyDisk(ctx context.Context, vmInfo compute.Virtual
 	return
 }
 
+// DestroyWindowsVMWithResources destroys the Windows VMs along with all the errors associated with it
+func (az *AzureProvider) DestroyWindowsVM(vmName string) error {
```

Comment:
nit: Should this function have a check to see if the VM actually exists? In the case it doesn't exist, return nil.